### PR TITLE
Fast Front-ends: Load FullStory Async

### PR DESCRIPTION
**Pull request by hamstu**
## Description

Loading FullStory async in the front-end instead of deciding server-side.

## Checklist

-   [ ] My code follows the code style and guidelines of this project. 
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [ ] I have tested different user stories. 
-   [ ] I have identified similar or related features and tested they are still working.
-   [ ] I have performed a self-review of my own code.
-   [ ] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [ ] I have commented my code, particularly in hard-to-understand areas.
-   [ ] I kept accessibility in mind by following the [a11y checklist.](https://www.notion.so/buffer/Workflow-Checklist-e64d86eb795140bcbfdc16d1c72e573f)
-   [ ] I have considered [security, abuse & compliance](https://www.notion.so/buffer/Engineering-Wiki-f34142d290304c35bebadf76cc9cc89e#cc6dcc7617184227b77da2e1b262a563) implications of my changes.

#### Staging Deployment

To visit the URL of the staging deployment, please click "Show all checks" at the bottom of this PR and click "details" next to `bufferbotbrains/cicd-buffer-publish-legacy`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -160,11 +161,11 @@
     "@storybook/addon-storyshots": "5.3.19",
     "@storybook/addons": "5.3.19",
     "@storybook/react": "5.3.19",
+    "@testing-library/cypress": "6.0.0",
     "@testing-library/dom": "^7.2.2",
     "@testing-library/jest-dom": "^5.5.0",
     "@testing-library/react": "^10.0.3",
     "@testing-library/user-event": "^10.1.0",
-    "@testing-library/cypress": "6.0.0",
```
Comment (anafilipadealmeida) [inline package.json:167]: 😂

File: packages/thirdParty/middleware.test.js [hunk 2]
```diff
@@ -1,12 +1,11 @@
 // /* global FS, Appcues */
 import { actionTypes as dataFetchActionTypes } from '@bufferapp/async-data-fetch';
+import * as FS from '@fullstory/browser';
```
Comment (anafilipadealmeida) [inline packages/thirdParty/middleware.test.js:3]: Would it make sense to also import it as `import * as FullStory`? I'm just thinking that sometimes having consistency in the name allows us to be more successful when using the search tool 🤔

**Review by anafilipadealmeida — APPROVED** [review]

File: packages/thirdParty/middleware.test.js (hunk 2 above)
Comment (hamstu) [reply packages/thirdParty/middleware.test.js:3]: I like that idea - thanks! 👍

File: packages/thirdParty/middleware.js [hunk 3]
```diff
@@ -59,20 +61,15 @@ export default ({ dispatch, getState }) => next => action => {
...
-              productFeatures: { planName },
-            } = getState();
-            if (planName !== 'free') {
-              window.FS.identify(id, {
-                pricingPlan_str: planName,
-              });
-            }
-          }
-        }
+      if (!action.result.isFreeUser && process.env.NODE_ENV === 'production') {
```
Comment (hamstu) [inline packages/thirdParty/middleware.js:64]: Webpack will replace `process.env.NODE_ENV` with the actual environment string at build time. (See the webpack config changes.)

**Pull request merged**